mobify/progressive-web-scaffold
# WEB-1233: Load main.js async as well

From the initial double fetch PR, main.js was not downloading until capture.js is done. main.js should download in parallel.

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1233
 **Linked PRs**:
https://github.com/mobify/crabtree-evelyn-progressive/pull/576

**Before**
![screen shot 2017-03-07 at 5 51 14 pm](https://cloud.githubusercontent.com/assets/4543395/23686541/158a6f0a-035f-11e7-9c80-39d8b814f54d.png)


**After**
![screen shot 2017-03-07 at 5 49 21 pm](https://cloud.githubusercontent.com/assets/4543395/23686544/19420676-035f-11e7-8075-bc9ec8a02062.png)


## Changes
- Extract the changes from Crabtree
  - wrap vendor, main, and capture in promises before starting up the app.

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Confirm in Chrome's network tab that main.js is downloading in parallel with capture.js and vendor.js
